app/presenters/tree_builder_policy_simulation.rb

```diff
@@ -0,0 +1,136 @@
+        nodes.push(:id       => node['id'],
+                   :text     => name.html_safe,
+                   :image    => icon,
+                   :tip      => node['description'],
+                   :policies => node['policies'])
+      end
+    end
+    if nodes.empty?
+      nodes.push(:id => nil, :text => _("Items out of scope"), :image => 'blank', :cfmeNoClick => true)
+    end
+    count_only_or_objects(count_only, nodes)
+  end
+
+  def skip_node(node)
+    value = true
+    if @policy_options[:out_of_scope] == true && node["result"] != "N/A"
+      value = false
+    elsif @policy_options[:passed] == true && node["result"] != "allow"
+      value = false
+    elsif @policy_options[:passed] == false && @policy_options[:failed] == true && node["result"] != "deny"
+      value = false
+    end
+    value
+  end
+
+  def policy_nodes(parent)
+    nodes = []
+    parent[:policies].sort_by { |a| a["description"] }.each do |node|
+      active_caption = node["active"] ? "" : _(" (Inactive)")
+      icon = node_icon(node["result"])
+      name = "<b>" + _("Policy%{caption}: ") % {:caption => active_caption} + "</b> #{node['description']}"
+      policy_node = {:id         => node['id'],
+                     :text       => name.html_safe,
+                     :image      => icon,
+                     :tip        => node['description'],
+                     :conditions => node['conditions']}
+      nodes.push(policy_node) unless skip_node(node)
+    end
+    nodes
+  end
+
+  def condition_node(parent)
+    nodes = []
+    parent[:conditions].sort_by { |a| a["description"] }.each do |node|
+      next if @policy_options[:out_of_scope] == false && [:conditions]["result"] == "N/A"
+      icon = node_icon(node["result"])
+      name = "<b>" + _("Condition: ") + "</b> #{node['description']}"
+      nodes.push(:id         => node['id'],
+                 :text       => name.html_safe,
+                 :image      => icon,
+                 :tip        => node['description'],
+                 :scope      => node['scope'],
+                 :expression => node["expression"])
+    end
+    nodes
+  end
+
+  def scope_node(parent)
+    icon = parent[:scope]["result"] ? "checkmark" : "na"
+    name, tip = exp_build_string(parent[:scope])
+    {:id => nil, :text => name.html_safe, :image => icon, :tip => tip.html_safe}
+  end
+
+  def expression_node(parent)
+    icon = parent[:expression]["result"] ? "checkmark" : "na"
+    name, tip = exp_build_string(parent[:expression])
+    {:id => nil, :text => name.html_safe, :image => icon, :tip => tip.html_safe}
+  end
+
+  def x_get_tree_hash_kids(parent, count_only)
+    nodes = []
+    nodes.concat(policy_nodes(parent)) unless parent[:policies].blank?
+    nodes.concat(condition_node(parent)) unless parent[:conditions].blank?
+    unless parent[:scope].blank?
+      unless @policy_options[:out_of_scope] == false && parent[:scope]["result"] == "N/A"
```

Comment:
and nested condition with 2 `unlesses`? please, unnest and make the condition positive